pkgs/stdenv/generic/setup.sh

```diff
@@ -3,6 +3,9 @@ set -o pipefail
 
 : ${outputs:=out}
 
+# If unset, assume the default hardening flags.
+: ${NIX_HARDENING_ENABLE="fortify stackprotector pic strictoverflow format relro bindnow"}
```

Comment:
Interesting, this means we can write e.g. `${NIX_STORE-}` instead of `${NIX_STORE:-}`.